keps/sig-node/3008-qos-class-resources/README.md

```diff
@@ -0,0 +1,2165 @@
+  - [~~Support for cgroups blockio~~](https://github.com/containerd/containerd/pull/5490)
+
+The design paradigm here is that the container runtime configures the QoS-class
+resources according to a given configuration file. Enforcement on containers is
+done via OCI. User interface is provided through pod and container annotations.
+
+Container runtimes will be updated to support the
+[CRI API extensions](#cri-api)
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `k8s.io/kubernetes/pkg/kubelet/kuberuntime`: `2022-06-13` - `66.8%`
+- `k8s.io/kubernetes/pkg/apis/core/validation/validation.go`: `2022-06-13` - `82.1%`
+- `k8s.io/kubernetes/pkg/scheduler`
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+Alpha: Existing integration tests for kubelet and kube-scheduler will be
+extended to cover QoS-class resources.
+
+Beta: Existing integration tests for affected components (e.g. scheduler, node
+status, quota) are extended to cover QoS-class resources.
+
+- <test>: <link to test coverage>
+
```

Comment:
Is it helpful to have a mock CRI implementation that pretends to run a container with various QoS assignments, but doesn't really do so?

Reply:
Yes, if such a mock/stub CRI server implementation exists

Reply:
@sftim is there smth that's open regarding this?

Reply:
I think it helps to copy that detail into the KEP text (ie, that a mock CRI implementation will be helpful).
This is OK to resolve whether we do or don't add that detail.

Reply:
K, I added a note in the KEP. It would be helpful if such a thing existed, I don't think there are many integration tests for kubelet.

> NOTE: having a mock implementation of CRI server would be helpful in writing
> integration tests for kubelet.

Will mark this as resolved...